### DISCUSSION TITLE
What size SSD to start with? (Because I can't add more)

### DISCUSSION THREAD

**Discussion by eduncan911** [Security-Onion-Solutions/securityonion#3365] [Unsupported Versions] [answered]
Trying to plan my server rebuilds, and the only box I have remaining for SO is going to be a shared box with pfSense.  LGA2011v1/v2 dual cpu 128GB beast.  It only has 2x onboard 1Gbps, so I'm adding a 2x 10Gbps SPF+ card (it only has 1 slot). 
 **Will be running two VMs on Proxmox for pfSense and Security Onion, _to start with_**.  This is a homelan.  But, it's shared with about 14 family members and friends with full-time VPN over 1Gbps up/down internet (on their mobiles and tablets).  Most likely will get a 2nd WAN soon (StarLink!).

The problem is this box only has 2x SATA ports.  So, I only get 1 shot at installing some SSDs in RAID1 (most likely mdadm).

I am finding it difficult to gauge how much HDD space is needed to run SO, considering its scaling-out abilities.  I have several other machines with dozens of TBs of storage I could "spread out SO over."

Overall, I'm trying to plan a lightweight VM install first with basic port mirroring from my 10Gbps switch and Suricata forwarded to SO, that will allow me to scale-out in the future as I need to (most likely for disk-space) by forwarding that data over to my K8s cluster.

I'm not looking for Months or Years of trending data.  A month should be fine.

Maybe a better set of questions are: 

* What features should I avoid and get away with 500GB or so for a month?
* What features are easily movable to a K8s cluster to scale out SO as needed?

I am not finding much info on the 2nd one.  But still digging through the videos and docs.

---

OT: This "pfSense and SO on the same box" is giving me other ideas I'd like to experiment with.  Like, using Suricata on SO to block traffic - via virtual NICs across the VMs? :)

**Comment by weslambert** [comment, marked as answer]
I'd only enable what you want/need.  Regardless of what you enable or don't run, the more throughput you are monitoring, the more data that will be generated.  You can prune this with BPF to filter out unwanted traffic (from interface/Zeek/Suricata/PCAP perspective), as well as filtering Zeek logs, or only enabling certain Zeek logs.  With regard to Suricata, you'll want to tune those rules as well, to ensure you aren't generating thousands of unnecessary alerts, honing in on what's most important to you.   You might also consider trimming/disabling Zeek's file extraction if you are trying to conserve space.

Space will primarily be consumed (aside from Docker images) by:

- PCAP in /nsm/pcap
- Zeek logs in /nsm/zeek/logs
- Suricata eve.json in /nsm/suricata
- ES data in /nsm/elasticsearch (combination of Suricata, Zeek, Wazuh, osquery, Windows logs, etc)
- Additional data sources like OSQ's results.log (if enabled)
